lib/engine/route.rb

```diff
@@ -257,11 +265,19 @@ def corporation
     end
 
     def connection_hexes
-      @connection_hexes ||= connections&.map(&:id)
+      @connection_hexes ||= if @train.local? && connection_data.one? && connection_data[0][:connection].paths.empty?
+                              [['local', connection_data[0][:left].hex.id]]
+                            else
+                              connections&.map(&:id)
+                            end
     end
 
     private
 
+    def add_node_connection(node)
```

Comment:
make this a bit more explicit

def add_single_node_connection
def add_solo_connection
etc